File: solutions/49. Group Anagrams/49.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        h = {}
        for s in strs:
            sorted_s = ''.join(sorted(s))
            if sorted_s in h:
                h[sorted_s].append(s)
            else:
                h[sorted_s] = [s]
        return h.values()

File: solutions/49. Group Anagrams/49.rb
# @param {String[]} strs
# @return {String[][]}
def group_anagrams(strs)
    h = {}
    strs.each do |s|
        sorted = s.chars.sort.join
        h[sorted] ? h[sorted] << s : h[sorted] = [s]
    end
    h.values
end

# Hint for a better solution in terms of perf:
# use the method `group_by` on strs and group by sorted string